courses/serializers.py

```diff
@@ -93,6 +93,15 @@ def get_description(self, instance):
         """Description"""
         return instance.page.description if instance.page else None
 
+    def get_courseruns(self, instance):
+        """Unexpired and unenrolled course runs"""
+        if hasattr(self.context.get("request"), "user"):
+            user = self.context["request"].user
+            active_runs = instance.available_runs(user)
+        else:
+            active_runs = instance.unexpired_runs
```

Comment:
Should we have this branch? I'm not sure we would ever use it since we would I think always have the user in the context for the course serializer.

Reply:
Besides unit tests, I wasn't sure if there were situations where that might happen so I left it in just in case.  I can remove it if you prefer.

Reply:
If I remove it, 16 unit tests fail because there is no request in the context.  I could modify `available_runs` so that user is an optional keyword, and if null returns the result of `unexpired_runs`, but that's just moving the branch from one place to another.

Reply:
Is there any use case for a serializer without the user context outside of unit tests? I am concerned that this could lead to subtle errors here if we leave out the user context at some point in the future. If there is no production use of this line then we should alter the unit tests to include the user context in all cases IMO